python/dask_cudf/dask_cudf/expr/_collection.py

```diff
@@ -71,6 +81,13 @@ def groupby(
                 f"`by` must be a column name or list of columns, got {by}."
             )
 
+        if as_index is not True:
+            raise NotImplementedError(
+                f"`as_index` is not supported by dask-expr. Please disable "
+                "query planning, or reset the index after aggregating.\n"
+                f"{_LEGACY_WORKAROUND}"
+            )
```

Comment:
*nit*: This is a somewhat confusing error message. The only way to get past it with query planning _enabled_ is to say `as_index=True`, but the error message seems to say "as_index=True` is not handled by dask-expr.

Do you mean:

> `dask-expr` only supports `as_index=True`. For `as_index=False` either disable query planning or reset the index with `reset_index` after aggregating.

WDYT?

Reply:
It sounds like dask-expr doesn't actually have support for the `as_index` keyword arg in general and always follows the behavior of `as_index=False`, so perhaps we should consider:

- checking if the kwarg is provided at all, emitting a `FutureWarning` if so
- if the kwarg isn't `as_index=True`, raise the error description suggested above

Reply:
The upstream `dask.dataframe` API has always raised an error when `as_index` is used, but dask-cudf has used a distinct groupby API until now.

I agree that the real problem is that `as_index` is not supported at all by dask-expr. Therefore @charlesbluca's suggestion is probably the most "correct". With that said, I'm feeling a bit hesitant to add more noise for something that technically "works fine" :/